src/devices/machine/1mb5.cpp

```diff
@@ -76,18 +76,22 @@ READ8_MEMBER(hp_1mb5_device::cpu_r)
 	case 1:
 		// Read IB
 		res = m_ib;
-		m_ibf = false;
-		update_halt();
+		if (m_ibf) {
+			machine().scheduler().boost_interleave(attotime::from_usec(50) , attotime::from_usec(100));
+			space.device().execute().spin();
```

Comment:
To me, this looks like an incredibly gross hack. There must be some better of way of handling the synchronization problem here.

Reply:
Well, it's surely an incredibly gross hack but on the other hand I'm not a wizard on scheduler manipulation. What is your suggestion for a more elegant hack, please? The point here is that HP developers relied heavily on race conditions when they were sure of the outcome in order to save a few "wait" instructions. So I tried to force MAME scheduler into some tight interleaving to achieve the same result as real processors executing in parallel. When the scheduler uses the normal time quantum a processor gets too far ahead of the one it's interacting with.

Reply:
Maybe you could try `config.set_perfect_quantum()` in the machine config?